### PR TITLE
Resource limiting: cargo supply, roundstart stockpile reductions (Armoury & Med) and limited sleepers.

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

 - Sleepers no longer product reagents, they inject from beakers stored within.
 - Sleeper upgrades allow for more beakers to be inserted, and improve the efficiency of injection.
 - The number of guns and armour in the armoury at roundstart has been reduced
 - The number of medical kits in medical storage at roundstart has been reduced
 - Cargo now has a stock value on each order, which increases over time (Every 2 minutes some stock will increase, with the items that have lower current stock compared to max stock being prioritised).
 - Moved some variables from SSshuttle to SSsupply
 - Roundstart budget amount doubled to ~6000 credits.
 - Sleepers will display the dominant chemical on the UI.
 - Sleeper will have a base efficiency of 1.25, which will go to 2.5 when fully upgraded. (Total amount of effective chems is multiplied by the efficiency)
 - The number of items in the armoury will scale with the number of members of security.
 - The number of items in roundstart medical storage will scale with the number of medical personnel. The medkits spawned will also be randomised.

## Why It's Good For The Game

The main intention of this PR is to increase the interactions between job roles.

Supply Changes:
 - Supply stock promotes longer term stock control rather than attempting to panic buy at the last minute. You can no longer ship 70 energy guns in a single order, but need to organise with cargo to purchase things to keep a reasonable stock.
 - Medical has signficantly less medkits roundstart. There is not enough to last the entire round, and they will need to purchase more through cargo if they want to stay stocked up.
 - Sleepers no longer produce free chemicals which can be used to get around needing to consume first aid kit resources. As a medical doctor, you will need to talk with the chemist to get chemicals for the sleepers, or work with cargo to get first aid kits to heal players.
 - The roundstart gear in the armoury has been reduced. Security will have to work with cargo to purchase weapons to stock up their armoury in the event that they need more gear. Depending on the map, the weapons in the armoury are different. More modern maps have a focus on energy weapons, while stations that are older lore-wise have less energy weapons but more ballistics (For example, kilo has 3 shotguns but only 1 energy and laser gun, while Corg has 2 SMGs but 4 energy guns).

These changes should promote longer term interactions with cargo rather than single binge purchases and make cargo a more vital part of the station in terms of normal operations, and extreme circumstances.

This PR could be broken down, however all the changes depend on each other to actually achieve my design goals, so I have included them for ease and to ensure they get merged together, even if it makes the review process slightly longer.

I'd like to make cargo be more exploitable for antag roles, however that can be done next weekend since its getting late now.

## Testing Photographs and Procedure

<details>

image

image

image

image

image

image

Double efficiency when fully upgraded.

image

image

image

image

image

image

After some time, the cargo stock has replenished:

image

</details>

## Changelog
:cl:
tweak: Cargo stock is now limited and now replenishes supply over time. You can no longer mass purchase large quanitites of the same item in a short period of time.
balance: Reduces amount of starting medkits in medical
balance: Reduces amount of starting weapons in armoury
tweak: Sleeper is no longer capable of producing chemicals. It needs beakers from chemistry in order to function. Sleepers will display the name of the dominant chem inside beakers, so you can get creative and hide poisons inside it.
tweak: Sleeper upgrades improve injection efficiency. (5 units used for every 10 units injected)
code: Moved some supply related variables away from SSshuttle and into the new supply subsystem.
balance: The station now starts with 50000 credits instead of 25000 distributed. This means that each budget will start with ~6000 credits instead of ~3000
/:cl:

### CODE REVIEW

**Comment by EvilDragonfiend** [comment]
I'd love to see the injection buttons are placed to right side rather than below. Moving buttond every time you click is annoying.

**Comment by EvilDragonfiend** [comment]
Did you check the roundstart sleepers only have default chems and manual construction ones don't have?

**Comment by PowerfulBacon** [comment]
Good point, you can create infinite chems by constructing and deconstructing a sleeper

**Comment by bluezorua** [comment]
question, will this mean you can have custom sleeper chem mixes?

**Comment by EvilDragonfiend** [comment]
> question, will this mean you can have custom sleeper chem mixes?

image

**Comment by Fronsis** [comment]
On the pictures provided as example on the Sleeper, those are the default values that start roundstart correct?

And for Armory i like the idea of having to work with Cargo early on the shift to secure more equipment for the armory, really hope the default budget of Defense can handle a few restocks because on shifts where cargo/explorers are not making money we'll be hit with the INSUFFICIENT FUNDS, the only ones who might have the upper hand on the armory state early on the shift are stealth-nukies but it's not like we have them every round so it's fine. 

As someone who's been playing a lot of sec/Warden lately i'm interested on how this is gonna play out on high-pop shifts

**Comment by PowerfulBacon** [comment]
Correct, the sleepers start with 30u bottles in them with whatever chems were set in the variables

File: code/game/machinery/Sleeper.dm [hunk 1]
```diff
@@ -14,26 +14,36 @@
...
 /obj/machinery/sleeper/Initialize(mapload)
 	. = ..()
 	occupant_typecache = GLOB.typecache_living
 	update_icon()
-	reset_chem_buttons()
+	//Create roundstart chems
+	for (var/datum/reagent/default_chem as() in roundstart_chems)
+		var/obj/item/reagent_containers/glass/bottle/beaker = new(null)
+		beaker.reagents.add_reagent(default_chem, beaker.volume)
+		inserted_vials += beaker
```
Comment (PowerfulBacon) [inline code/game/machinery/Sleeper.dm]:
```suggestion
	if (mapload)
		for (var/datum/reagent/default_chem as() in roundstart_chems)
			var/obj/item/reagent_containers/glass/bottle/beaker = new(null)
			beaker.reagents.add_reagent(default_chem, beaker.volume)
			inserted_vials += beaker
```

**Comment by kit-katz** [comment]
update economica is a go

**Comment by mystery3525** [comment]
This amount of changes in this deserves to be split into separate PRs. Bunching sleeper nerfs with cargo re-balances does not feel right (Additionally the +2000 -2000 line changes), as there are multiple different changes that should have community input considered. Additionally, can the Photo evidence be placed into a collapsible tab to not make it take up two thirds of the scroll length?

**Comment by Penwin0** [comment]
Screenshot_1311

Is this intentional? (Box station virology)

**Comment by PowerfulBacon** [comment]
No, probably a poorly var-editted APC which got reset by the map editor. Will fix

**Comment by EvilDragonfiend** [comment]
Is there still 20u maximum per part tier? I think it is unnecesary at that point.

**Comment by FortSumter** [comment]
Can you put ANY chemical in the new sleepers? I can think of some funny shenanigans if you could.

**Comment by PowerfulBacon** [comment]
Yes, you can. However it displays the name of the dominant chemical in the solution on the button.

**Comment by EvilDragonfiend** [comment]
How would emag sleeper become?

**Comment by FortSumter** [comment]
Yes yes. We need more malpractice and sabotage that doesn't require funny antag gear.

**Comment by PowerfulBacon** [comment]
> How would emag sleeper become?

Pressing a button injects a random chemical

**Comment by Penwin0** [comment]
Needs merge conflicts fixed and map looked at, code looks good though

**Comment by Kenionatus** [comment]
Why not just display the beaker label. That would require either porting renaming beakers with pens or putting a hand labeller in med at round start. (At least on NSV, hand labellers work on beakers, so they probably do on bee too.)

Another suggestion from me would be to have multiple buttons in the interface of the sleeper, one for each inserted beaker.

(As you may suspect from my comment about NSV, I only play and code there, so feel free to disregard.)

**Comment by PowerfulBacon** [comment]
Originally I used the name of beakers, however it turns out that beakers can't be renamed with a pen, so I just opted to use the dominant chemical of a mix instead

**Comment by PowerfulBacon** [comment]
Bottles in vending machines have become unnamed.

**Comment by EvilDragonfiend** [comment]
> Bottles in vending machines have become unnamed. It would be nice if you could control the injection amount.

use `label_name` variable instead. I updated containers for your PR.

**Comment by EvilDragonfiend** [comment]
> Bottles in vending machines have become unnamed.

It's my fault. I didn't expect vanding machine uses `name` variable to track names.
I updated containers to use `label_name` instead. I will fix it soon.

**Comment by PowerfulBacon** [comment]
Why are there 140k diffs all of a sudden

**Comment by itsmeow** [comment]
merge conflict

**Comment by Tsar-Salat** [comment]
whattheshit1
what

**Comment by wikimody** [comment]
Low pop will be hit hard by this PR

**Comment by wikimody** [comment]
Ok just finished nukie round with this PR and it higligts the bigest problem it has with suplies. I was qutermater i had the money but i coudnt buy us more weponsto figt the nukies. It wasnt a qustion of waiting for stocs to refresh since we didnt have time to wait or time to prepeare by buing wepons before nukies were known. We won anyway but thats mostly because nukies were flukies this round.

**Comment by Fronsis** [comment]
As of last round 42787 we had funny space hole in armory,  unsure if this was done just now or the fix hasn't been pushed

But yeah so far it's uh.. interesting to see the lack of weapons, we did buy some on cargo and the little extra budget helped, though medbay was quite empty kit-wise despite having quite a lot of docs

Haven't played medical myself to comment on sleeper stuff but from the outside it was like ''damn they have no kits'', which upon arresting a couple of antags i realized they quickly hoarded all of em for their own use.

**Comment by PowerfulBacon** [comment]
The mass ordering in particular targets nukies and blobs. There should be enough weapons to arm the crew with a variety of weapons, however the weapons will be limited to whatever is in stock, meaning that not everyone can have a shotgun or the meta weapons, a lot of people will need to get energy weapons, laser guns, WTs or whatever is available.

Armoury hole has been fixed.

Changes to how medical people act with their medkits is interesting and something I want to continue looking into, I have seen them get quite a few kits but because of the randomness they can all be less used types.
My main wonder is are doctors actually using the ordering console to purchase more medkits?

**Comment by wikimody** [comment]
Then can i ask for you to make sure that each wepon type has at least 1 stock at the start and refreshes before 20 minute mark? That way you can prepeare beter against war nukies but you still canot spam wepons against blobs and stelf nukies. The whole point of war nukies is to make them much more powerfull at the cost of crew having time to prepare so this change woudl make sure crew can prepare. Otherwise you migt hapen upon a round where nukies decleared war and you cannot buy wepons for the station at all (not: I cannot have combat shotgun so I will take riot, but: I do not have anything that shots)

**Comment by MarkusLarsson421** [comment]
Recently played a game with the new sleeper and medkit changes as a chemist I absolutely **love** these changes!

However, there are some issues. Namely the previously discussed naming of chemicals. We do have 200 unit bags in the Chem Master. They can already be named inside the Chem Master. Why not use it? Could perhaps even have a dedicated sleeper only with blood bags then. I am thinking of my Perfluorodecalin mix where I use 1 unit of Perfluorodecalin and 4 units of Antitoxin (for the user to come out with 0 toxin damage) to heal 100 suffocation damage. But, the current system will just show it as "Antitoxin" and ignore everything else making it confusing to use for doctors. 

You might argue "Just use a Perfluorodecalin bag and an Antitoxin bag" but 10 units of Perfluorodecalin is overkill and a waste. Not to mention that there is a limit of six slots in the sleeper currently and this would take two slots instead of one.

Additionally, there might be an issue when it comes to the access of it. Currently anyone can add and remove everything in it. And as they are stored in beakers, 300 units of something annoyingly, difficult and painfully made can be taken out and thrown very easily. And if custom names were to be implemented, then anyone can store anything that isn't particularly... healthy under an innocent name. I'd like to see them being locked, whether to only Chemists and the CMO or all of the Medbay personnel.

Lastly, I think the Emag could be buffed in regard to the sleeper. As randomizing the button pressed is a bit stale, "Whoops, you got Bicaridine instead of Kelotane! Better deconstruct and construct it again!". If the sleeper would also be locked then I think Emag would give you access to it as well as letting you select one of the existing reagents in it. Of once pressed will fill the user with as much as possible of said reagent. 

So with all these suggestions it would enable, for example, a Syndicate assistant could get themselves inside Medbay, Emag to access it and put in a bag named "Salicyclic Acid" with 200 unints of Sulfonal. The next victim to have Salicyclic Acid used on them will then be injected a great amount of Sulfonal, whether the entire bag, a large amount or simply 10 units could be discussed.

**Comment by EvilDragonfiend** [comment]
My opinion for emagged sleeper is that it doesn't show 'toxin reagent list'
It will show full reagent list in a patient, but just without toxin. That'll be easy to use.

**Review by PestoVerde322 — APPROVED** [review]

**Pull request merged**